Update the `log` to `filestream` migration guide (#43641)
elastic/beats: docs/reference/filebeat/migrate-to-filestream.md

/var/log/my-old-files1.json (0%)
```

The progress of data collection is shown for each file.

After this migration we expect that the following files will continue to be ingested by the new `filestream` inputs from their current positions:

```
/var/log/java-exceptions3.log (75%)
/var/log/java-exceptions4.log (0%)
/var/log/java-exceptions5.log (0%)
/var/log/my-application2.json (5%)
/var/log/my-application3.json (0%)
/var/log/my-old-files1.json (0%)
```

## Replacing with `filestream`

::::{important}
Don't start Filebeat until you finish steps 1-3 of this guide. Otherwise, you'll see invalid data or data duplication.
The intermediate configuration snippets are for illustration purposes only.
::::

### Step 1: Unique identifiers [step_1]

> Every `filestream` input **must** have a unique identifier. We suggest creating meaningful identifiers for each new `filestream` input. This makes it easier to troubleshoot if something goes wrong.

Reviewer: ```suggestion
Every `filestream` input **must** have a unique identifier as `id`. Using meaningful identifiers for each new `filestream` input will also make it easier to troubleshoot if something goes wrong.
```

Enforcing the "must"

```yaml
filebeat.inputs:
 - type: filestream
   enabled: true
   id: my-java-collector
   take_over:
     enabled: true
   paths:
     - /var/log/java-exceptions*.log

 - type: filestream
   enabled: true
   id: my-container-input
   take_over:
     enabled: true
   paths:
     - /var/lib/docker/containers/*.log

 - type: filestream
   enabled: true
   id: my-application-input
   take_over:
     enabled: true
   paths:
     - /var/log/my-application*.json

 - type: filestream
   enabled: true
   id: my-old-files
   take_over:
     enabled: true
   paths:
     - /var/log/my-old-files*.log
```

### Step 3: Migrate additional parameters if necessary [step_3]

Some configuration options are renamed or moved in `filestream`:

| **`log` input** | **`filestream` input** |
| --- | --- |
| recursive_glob.enabled | [prospector.scanner.recursive_glob](/reference/filebeat/filebeat-input-filestream.md#filestream-recursive-glob) |
| harvester_buffer_size | [buffer_size](/reference/filebeat/filebeat-input-filestream.md#_buffer_size) |
| max_bytes | [message_max_bytes](/reference/filebeat/filebeat-input-filestream.md#_message_max_bytes) |

> | json | [parsers.n.ndjson](/reference/filebeat/filebeat-input-filestream.md#filebeat-input-filestream-ndjson) |

Author: yes, it signifies the list and the item's index.
